### PR TITLE
emacs: add customization bindings

### PR DESCRIPTION
**Pull request by gleon99**

### CODE REVIEW

File: emacs.d/init.el [hunk 1]
```diff
@@ -16,6 +16,10 @@
       package-archives '(("gnu" . "https://elpa.gnu.org/packages/")
                          ("melpa" . "https://melpa.org/packages/")))
 
+(defgroup lgenkin nil
```
Comment (drrlvn) [inline emacs.d/init.el]: Group should have a generic name, like `my/customizations`.

File: emacs.d/init.el [hunk 2]
```diff
@@ -843,4 +857,24 @@ _M-p_: Unmark  _M-n_: Unmark  _q_: Quit"
   (unbind-key "<tab>" yas-minor-mode-map)
   (yas-global-mode 1))
 
+(defun my/scroll-up (n)
```
Comment (drrlvn) [inline emacs.d/init.el]: Functions should go in `config-defuns.el` and be autoloaded.

File: emacs.d/init.el [hunk 3]
```diff
@@ -843,4 +857,24 @@ _M-p_: Unmark  _M-n_: Unmark  _q_: Quit"
...
+     "Scroll up N lines."
+     (interactive "p")
+     (scroll-up n))
+
+(defun my/scroll-down (n)
+  "Scroll down N lines."
+  (interactive "p")
+  (scroll-down n))
+
+(when scroll-bindings
```
Comment (drrlvn) [inline emacs.d/init.el]: Bindings should be together with all other bindings.

File: emacs.d/init.el [hunk 4]
```diff
@@ -16,6 +16,10 @@
       package-archives '(("gnu" . "https://elpa.gnu.org/packages/")
                          ("melpa" . "https://melpa.org/packages/")))
 
+(defgroup customizations nil
```
Comment (drrlvn) [inline emacs.d/init.el]: Use `my/` prefix. We use that as a namespace replacement.

File: emacs.d/init.el [hunk 5]
```diff
@@ -41,6 +45,17 @@
...
+            "Line scroll bindings"
+            :type 'boolean
+            :group 'customizations)
+ 
+(defcustom window-selection-bindings nil
+            "Window selection bindings"
+            :type 'boolean
+            :group 'customizations)
+
+
```
Comment (drrlvn) [inline emacs.d/init.el]: Too many blank lines.

File: emacs.d/lisp/config-defuns.el [hunk 6]
```diff
@@ -452,6 +452,19 @@ COUNT are set in the same way as the original function."
...
+  (interactive "p")
+  (scroll-up n))
+
+;;;###autoload
+(defun my/scroll-down (n)
+  "Scroll down N lines."
+  (interactive "p")
+  (scroll-down n))
+
+
```
Comment (drrlvn) [inline emacs.d/lisp/config-defuns.el]: Too many blank lines.

File: emacs.d/init.el [hunk 7]
```diff
@@ -41,6 +45,17 @@
 
 (defconst rg-executable (executable-find "rg"))
 
+(defcustom scroll-bindings nil
```
Comment (drrlvn) [inline emacs.d/init.el]: Use `my/` prefix for variables as well.

File: emacs.d/lisp/config-defuns.el (hunk 6 above)
Comment (gleon99) [reply emacs.d/lisp/config-defuns.el]: Fixed

File: emacs.d/init.el (hunk 7 above)
Comment (gleon99) [reply emacs.d/init.el]: Fixed

File: emacs.d/init.el (hunk 4 above)
Comment (gleon99) [reply emacs.d/init.el]: Fixed

Comment (gleon99) [reply emacs.d/init.el]: Though, I'm not quite sure about the logic behind it..

It looks a bit weird, don't you think?
image

File: emacs.d/init.el [hunk 8]
```diff
@@ -87,6 +101,13 @@
 
 (bind-key [remap goto-line] #'my/goto-line-with-feedback)
 
+(when my/scroll-bindings
+  (bind-key "C-e" #'my/scroll-up)
+  (bind-key "C-y" #'my/scroll-down))
+
+(loop for i from 1 to 9 do
```
Comment (drrlvn) [inline emacs.d/init.el]: Use `(dotimes (i 10) ...)`.

**Pull request merged**